demos/link-prediction/random-walks/ctdne-link-prediction.ipynb

```diff
@@ -0,0 +1,680 @@
+    "from sklearn.manifold import TSNE\n",
+    "from stellargraph import StellarGraph\n",
+    "\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load graph data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def load_enron(data_dir):\n",
+    "    edges_path = os.path.join(data_dir, \"ia-enron-employees.edges\")\n",
+    "    edges = pd.read_csv(edges_path, sep=\" \", header=None)\n",
+    "\n",
+    "    # clean up columns\n",
+    "    edges.columns = [\"source\", \"target\", \"x\", \"time\"]\n",
+    "    edges[[\"source\", \"target\"]] = edges[[\"source\", \"target\"]].astype(str)\n",
+    "    edges = edges.drop(columns=[\"x\"])  # unused column\n",
+    "\n",
+    "    # time in seconds\n",
+    "    edges[\"time\"] = edges[\"time\"]\n",
+    "    nodes = pd.DataFrame(\n",
+    "        np.unique(pd.concat([edges[\"source\"], edges[\"target\"]], ignore_index=True)),\n",
+    "        columns=[\"id\"],\n",
+    "    ).set_index(\"id\")\n",
+    "\n",
+    "    return nodes, edges"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nodes, edges = load_enron(data_dir)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Split edges\n",
+    "\n",
+    "We want to split the edges into parts that can be used for our link prediction task:\n",
+    "* the oldest edges are used to create the graph structure\n",
+    "* the recent edges are what we are interested in predicting - we randomly split this part further into training and test sets."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of edges in graph: 37929\n",
+      "Number of edges in training set: 9524\n",
+      "Number Of edges in test set: 3119\n"
+     ]
+    }
+   ],
+   "source": [
+    "# subset of edges to split\n",
+    "train_subset = 0.25\n",
+    "test_subset = 0.25\n",
+    "\n",
+    "# number of edges to be kept in the graph\n",
+    "num_edges_graph = int(len(edges) * (1 - train_subset))\n",
+    "\n",
+    "# keep older edges in graph, and predict more recent edges\n",
+    "edges_graph = edges[:num_edges_graph]\n",
+    "edges_other = edges[num_edges_graph:]\n",
+    "\n",
+    "# split recent edges further to train and test sets\n",
+    "test_mask = np.random.rand(len(edges_other)) < test_subset\n",
+    "edges_train = edges_other[~test_mask]\n",
+    "edges_test = edges_other[test_mask]\n",
+    "\n",
+    "print(\n",
+    "    f\"Number of edges in graph: {len(edges_graph)}\\n\"\n",
+    "    f\"Number of edges in training set: {len(edges_train)}\\n\"\n",
+    "    f\"Number Of edges in test set: {len(edges_test)}\"\n",
+    ")"
```

Comment:
I think this cell (and the `create_link_examples` one too) can theoretically use [`EdgeSplitter`](https://stellargraph.readthedocs.io/en/latest/api.html#stellargraph.data.EdgeSplitter), and, even if not, can use [`sklearn.model_selection.train_test_split`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html) to be a bit more declarative. What do you think?

Reply:
Hmm right, I think the initial split still needs to be done via grabbing slices of the pandas dataframe since we want to split the old vs new edges. I'll switch the train/test split to use one of those though